### PR TITLE
Refactor existing github actions, add quality workflow

### PR DESCRIPTION
**Pull request by collindutter**
#124 
Runs `pylint` on the files that have changed in the PR.

### CODE REVIEW

File: griptape/memory/tool/base_tool_memory.py
Comment (collindutter) [inline griptape/memory/tool/base_tool_memory.py:1]: Updated a minor file to make eslint happy.

**Review by vasinov — APPROVED** [review]

**Comment by vasinov** [comment]
@collindutter, do you have any ideas as to why this can't be merged?

**Comment by collindutter** [comment]
@vasinov not sure, I just rebased onto `main` just in case though.

**Comment by vasinov** [comment]
Seems to have worked—merging! :)

**Pull request merged**